src/Raven.Studio/typescript/components/pages/resources/manageServer/certificates/partials/authEnabled/CertificatesReplaceServerModal.tsx

```diff
@@ -0,0 +1,135 @@
+import classNames from "classnames";
+import Modal from "components/common/Modal";
+
+export default function CertificatesReplaceServerModal() {
+    const dispatch = useAppDispatch();
+    const { manageServerService } = useServices();
+    const { reportEvent } = useEventsCollector();
+
+    const form = useForm<FormData>({
+        resolver: yupResolver(schema),
+        defaultValues: {
+            certificateAsBase64: "",
+            certificatePassphrase: "",
+            isReplaceImmediately: false,
+        },
+    });
+
+    const { control, formState, handleSubmit, reset } = form;
+
+    const formValues = useWatch({ control });
+
+    const asyncGetClusterDomains = useAsync(manageServerService.getClusterDomains, []);
+
+    const handleReplace: SubmitHandler<FormData> = async (formData) => {
+        return tryHandleSubmit(async () => {
+            reportEvent("certificates", "replace");
+            await manageServerService.replaceClusterCertificate(
+                certificatesUtils.mapReplaceServerToDto(formData),
+                formData.isReplaceImmediately
+            );
+            reset(formData);
+            dispatch(certificatesActions.fetchData());
+            dispatch(certificatesActions.isReplaceServerModalOpenToggled());
+        });
+    };
+
+    return (
+        <Modal show size="lg" centered contentClassName="modal-border bulge-warning">
+            <FormProvider {...form}>
+                <form onSubmit={handleSubmit(handleReplace)}>
+                    <Modal.Header
+                        className="vstack gap-4"
+                        onCloseClick={() => dispatch(certificatesActions.isReplaceServerModalOpenToggled())}
+                    >
+                        <div className="text-center">
+                            <Icon icon="refresh" className="fs-1" color="warning" margin="m-0" />
+                        </div>
+                        <div className="text-center lead">Replace server certificates (cluster-wide)</div>
+                    </Modal.Header>
+                    <Modal.Body>
+                        <LazyLoad active={asyncGetClusterDomains.loading}>
+                            <RichAlert title="" variant="info" className="my-2">
+                                <span>
+                                    Replace all server certificates in the cluster without shutting down the servers.
+                                    The update will happen when all nodes of the cluster confirm the replacement, or
+                                    when there are 3 days left for expiration.
+                                </span>
+                                <div className={classNames(asyncGetClusterDomains.result?.length ? "" : "d-none")}>
+                                    Please verify that the new certificate contains all of the following domain names in
+                                    the CN or ASN properties of the certificate:
```

Comment:
Did you mean SAN (Subject Alternative Names) ? Also using CN is deprecated

> Subject field (or CN attribute) is deprecated by [RFC 2818](https://datatracker.ietf.org/doc/html/rfc2818). Instead, all > relevant names must be specified in Subject Alternative Names (SAN) certificate extension. You can populate this > extension under Subject field editor in certificate request manager.

Reply:
IDK, copied from old certificates view